### ISSUE TITLE
Inject service instead of service container

### ISSUE THREAD

**Issue by qferr** [APY/APYDataGridBundle#790]
The `Grid` should not be dependent on the service container, only on the services it needs. The service can be used in an environment which is not using the Symfony service container. The code would be easier to read, maintain and unit test.

**Comment by clubdesarrolladores** [comment]
The bigger dependency is the session component?, its possible rewrite the grid to implement a "Post Redirect Get pattern". Post take the filter's parameters, and redirect with these filters in the query string, later a GET controller make the sql query and renderize the template.
The paginator and header columns order, can use the GET controller directly. No session is needed.

The grid class has more than 2000 lines and too many responsabilities, we must explode this class in several classes (and make these testables!).

+1 for this issue.

**Comment by qferr** [comment]
I agree, for which reason I developed a `Grid::handleRequest` (like symfony form component) method for manage filters and sorts. For now, it just calls the method isReadyForRedirect but must be rewritten. To reduce the coupling, we could handle requests in a `RequestHandler` object for example.

**Comment by ekkinox** [comment]
+1

**Comment by giovkanata** [comment]
+1

**Comment by hkulekci** [comment]
+1
